File: AquaTrack/src/main/java/com/aquatrack/service/WaterUsageService.java
package com.aquatrack.service;

import com.aquatrack.dto.waterusage.CreateWaterUsageRequestDto;
import com.aquatrack.dto.waterusage.UpdateWaterUsageRequestDto;
import com.aquatrack.dto.waterusage.WaterUsageResponseDto;

import java.util.List;

public interface WaterUsageService {

    /**
     * Create a new water usage log.
     */
    WaterUsageResponseDto createWaterUsage(
            Long householdId,
            CreateWaterUsageRequestDto requestDto
    );

    /**
     * Update an existing water usage log.
     */
    WaterUsageResponseDto updateWaterUsage(
            Long householdId,
            Long waterUsageId,
            UpdateWaterUsageRequestDto requestDto
    );

    /**
     * Get a water usage log by Id.
     */
    WaterUsageResponseDto getWaterUsageById(
            Long householdId,
            Long waterUsageId
    );

    /**
     * Get all water usage logs of a household.
     */
    List<WaterUsageResponseDto> getAllWaterUsage(
            Long householdId
    );

    /**
     * Delete a water usage log.
     */
    void deleteWaterUsage(
            Long householdId,
            Long waterUsageId
    );

}

File: Aquatrack-frontend/src/services/waterUsageService.js
import api from "../api/axios";

const BASE_URL = "/manager";

const waterUsageService = {

    // ==========================================
    // Get All Water Usage
    // ==========================================

    getAllWaterUsage: async (householdId) => {

        const response = await api.get(
            `${BASE_URL}/households/${householdId}/water-usage`
        );

        return response.data;

    },

    // ==========================================
    // Get Water Usage By Id
    // ==========================================

    getWaterUsageById: async (
        householdId,
        waterUsageId
    ) => {

        const response = await api.get(
            `${BASE_URL}/households/${householdId}/water-usage/${waterUsageId}`
        );

        return response.data;

    },

    // ==========================================
    // Create Water Usage
    // ==========================================

    createWaterUsage: async (
        householdId,
        data
    ) => {

        const response = await api.post(
            `${BASE_URL}/households/${householdId}/water-usage`,
            data
        );

        return response.data;

    },

    // ==========================================
    // Update Water Usage
    // ==========================================

    updateWaterUsage: async (
        householdId,
        waterUsageId,
        data
    ) => {

        const response = await api.put(
            `${BASE_URL}/households/${householdId}/water-usage/${waterUsageId}`,
            data
        );

        return response.data;

    },

    // ==========================================
    // Delete Water Usage
    // ==========================================

    deleteWaterUsage: async (
        householdId,
        waterUsageId
    ) => {

        const response = await api.delete(
            `${BASE_URL}/households/${householdId}/water-usage/${waterUsageId}`
        );

        return response.data;

    }

};

export default waterUsageService;